### ISSUE TITLE
migrate from docker-compse.yml files to docker python library

### ISSUE THREAD

**Issue by leoank** [intermine/intermine_boot#2]
## Problem
Intermine_boot uses `docker-compose.yml` files from [docker-intermine-gradle](https://github.com/intermine/docker-intermine-gradle) repo for launching containers. Intermine_boot clones `docker-intermine-gradle` repo for the files and then open up a shell to execute docker-compose commands.
This approach works, but we cannot guarantee predictable behaviour with this. What if files in the `docker-intermine-gradle` changes? Also, we have to clone entire `docker-intermine-gradle` for just two files. 

## Outline of a Solution
We want to use [docker-py](https://github.com/docker/docker-py.git) library.
Translate `docker-compose.yml` into python functions using `docker-py` library.
Then replace the old way of starting containers with these new functions.

**Comment by brite107** [comment]
I’d like to take this issue :)

**Comment by Mightynasty** [comment]
I cant take this one

**Comment by heralden** [comment]
Hello all! Due to there being a lot of interest in this project compared to the amount of issues, we've decided to not assign issues and allow anyone interested to submit a PR. Obviously we won't be able to merge all the PRs, so don't worry about being first. All PRs whether merged or not, will be part of our evaluation of an intern.

It would also be great if you could explain your approach and rationale in the PR description. This only needs to be as long or as in depth as you feel it needs to be for the specific task.

In summary, just go ahead and start working on this if you want!

**Comment by kathivashkiv** [comment]
I'd like to work on this issue too

**Comment by Mightynasty** [comment]
@uosl Hi, a question! When trasnferring .yml to py docker, should all the environment variables be preserved? And also, should the docker.py file changed accordingly?

**Comment by heralden** [comment]
@Mightynasty You can start with preserving only the UID and GID environment variables which are currently passed to docker-compose in docker.py, using the equivalent py-docker functionality instead. We will probably add support for the remaining ones to be passed through as CLI arguments in the future, but you don't need to handle these now.
You can make the changes in the docker.py file or create a new file depending on what you find most suitable.

**Comment by 22PoojaGaur** [comment]
@uosl should we close this issue?

**Comment by leoank** [comment]
yup!
